Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+RemoteExecutor is a simple console application which accepts arguments that point to an existing method in the test assembly and additional arguments you might need in your test. For additional information see https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs and https://xunit.github.io/docs/running-tests-in-parallel.html
+
+Example (skipping additional usings):
+```cs
+using System.Diagnostics;
+
+public class HttpWebRequestTest : RemoteExecutorTestBase
+{
+    [Fact]
+    public void DefaultMaximumResponseHeadersLength_SetAndGetLength_ValuesMatch()
+    {
+        RemoteInvoke(() =>
+        {
+            const int NewDefaultMaximumResponseHeadersLength = 255;
+            HttpWebRequest.DefaultMaximumResponseHeadersLength = NewDefaultMaximumResponseHeadersLength;
+            Assert.Equal(NewDefaultMaximumResponseHeadersLength, HttpWebRequest.DefaultMaximumResponseHeadersLength);
+
+            return SuccessExitCode;
+        }).Dispose();
+    }
+}
+```
+
+ # LoopbackServer
+When writing network related tests we try to avoid running tests against a remote endpoint if possible. We provide simple APIs to create a LoopbackServer and send responses. A high number of scenarios can be tested with it. For additional information see https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Net/Http/LoopbackServer.cs
+
+Example (skipping additional usings and class scoping):
+```cs
+using System.Net.Test.Common;
+
+[Fact]
+public async Task Headers_SetAfterRequestSubmitted_ThrowsInvalidOperationException()
+{
+    await LoopbackServer.CreateServerAsync(async (server, uri) =>
+    {
+        HttpWebRequest request = WebRequest.CreateHttp(uri);
+        Task<WebResponse> getResponse = request.GetResponseAsync();
+        await LoopbackServer.ReadRequestAndSendResponseAsync(server);
+        using (WebResponse response = await getResponse)
+        {
+            Assert.Throws<InvalidOperationException>(() => request.AutomaticDecompression = DecompressionMethods.Deflate);
+        }
+    });
+}
+```
+
+# Outerloop
+This one is fairly simple but often used incorrectly. When running tests which depend on outside influences like e.g. Hardware (Internet, SerialPort, ...) and you can't mitigate these dependencies, you might consider using the `[Outerloop]` attribute for your test. 
+With this attribute, tests are executed in a dedicated CI loop and won't break the default CI loops which get created when you submit a PR.
+To run Outerloop tests locally you need to set the msbuild property "Outerloop" to true: `/p:Outerloop=true`.
+To run Outerloop tests in CI you need to mention dotnet-bot and tell him which tests you want to run. See `@dotnet-bot help` for the exact loop names.
+
+This doesn't mean that you should mark every test which executes against a remote endpoint as Outerloop. See below.
```

Comment:
I think the current policy is that we _should_ mark any networking test as Outerloop. 
All of them are and should be currently marked as `[OuterLoop] // TODO: Issue #11345`. Until we can figure out a consistent way to try out multi-machine testing, something I've already started working on at https://github.com/dotnet/corefx/tree/master/src/Common/tests/System/Net/Prerequisites